swap_meet/vendor.py

```diff
@@ -1,2 +1,113 @@
 class Vendor:
-    pass
\ No newline at end of file
+    """
+    A class to represent a vendor, managing the inventory and providing
+    methods for adding, removing, and swapping items between vendors.
+    """
+
+    def __init__(self, inventory=None):
+        """
+        Initializes a Vendor instance with an optional inventory list.
+        If no inventory is provided, an empty list is initialized.
+        """
+        self.inventory = inventory or []
+
+    def add(self, new_item_added):
+        """
+        Adds the item to the `inventory`
+        and returns item
+        """
+        self.inventory.append(new_item_added)
+        return new_item_added
+
+    def remove(self, item_to_remove):
+        """
+        Removes the specified item from the `inventory` if it exists and returns item
+        """
+        if item_to_remove not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item_to_remove)
+            return item_to_remove
```

Comment:
Since you've reversed the logic to put the error checking first as a "guard clause" this allows us to unindent the main logic and remove `else`.

```python
if item_to_remove not in self.inventory:
    return False 

self.inventory.remove(item) 
return item
```

Prefer to have fewer levels of nested logic where possible.